### PR TITLE
Basic support for iOS, fix missing Rust lib

### PR DESCRIPTION
**Pull request by karniv00l**
Related to: https://github.com/KRTirtho/spotube/discussions/173

Only tested on simulator right now.

Screenshot 2022-10-13 at 19 44 50

Issues:

- download doesn't work (at least in simulator)
- ~playback bar covers bottom menu~
- super weird scrolling / swiping behavior (up/down) - practically unusable (at least in the simulator), I guess this is the change with the Flutter's ios/macos drag physics

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -52,7 +52,10 @@ dependencies:
   auto_size_text: ^3.0.0
   badges: ^2.0.3
   mime: ^1.0.2
-  metadata_god: ^0.2.0
+  metadata_god:
+    git:
+      url: https://github.com/KRTirtho/metadata_god.git
+      ref: 7d195fdde324b382fc12067c56391285807e6233
```
Comment (karniv00l) [inline pubspec.yaml:58]: this is the last working commit

Comment (KRTirtho) [reply pubspec.yaml:58]: In this commit I guess MacOS wasn't working. Does the MacOS application work?

**Comment by KRTirtho** [comment]
> playback overlay covers bottom menu

It is fixed now. Thanks to the new `ShellRoute` API of go_router. Now it's a part of the `bottomNavigationBar`. Both will show everywhere except Login Page

File: pubspec.yaml (hunk 1 above)
Comment (karniv00l) [reply pubspec.yaml:58]: This fixes the library on macOS, after this commit it breaks (it can't find the compiled library anymore)

Comment (KRTirtho) [reply pubspec.yaml:58]: Ah.. my bad. It's actually the latest commit :sweat_smile:

**Comment by karniv00l** [comment]
> It is fixed now. Thanks to the new `ShellRoute` API of go_router. Now it's a part of the `bottomNavigationBar`. Both will show everywhere except Login Page and PlayerView

rebased

**Comment by KRTirtho** [comment]
Thanks man. Great job ❤️🚀

**Pull request merged**